src/main/java/growthcraft/milk/common/tileentity/TileEntityButterChurn.java

```diff
@@ -217,26 +193,22 @@ public boolean tryTakeItem(IItemOperable.Action action, EntityPlayer player, Ite
     @TileEventHandler(event = TileEventHandler.EventType.NBT_READ)
     public void readFromNBT_ButterChurn(NBTTagCompound nbt) {
         this.shaftState = nbt.getInteger("shaft_state");
-        this.churns = nbt.getInteger("churns");
```

Comment:
You should support the recovery of this old churns variable into the new Churn device.

i.e. You should be reloading the state into the new device at least, unless it's already handled by the device in question, in which case, ignore my comment

Reply:
the number of churns is handled in the device. I used the ProgressiveDevice class with the number of churns as the time variable.